### PR TITLE
Ceph S3 tests % update

### PR DESCRIPTION
**Pull request by romayalon**
### Explain the changes
1. changed the replace of % to %25 to a more generic solution.

### Issues: Fixed #xxx / Gap #xxx
1. 

### Testing Instructions:
1.

### CODE REVIEW

**Review by nimrod-becker — APPROVED** [review]

File: src/endpoint/s3/s3_rest.js [hunk 1]
```diff
@@ -315,7 +315,7 @@ function get_bucket_and_key(req) {
     return {
         bucket,
         // decode and replace hadoop _$folder$ in key
-        key: decodeURIComponent(key.replace('%', '%25')).replace(/_\$folder\$/, '/'),
+        key: decodeURIComponent(encodeURIComponent(key)).replace(/_\$folder\$/, '/'),
```
Comment (guymguym) [inline src/endpoint/s3/s3_rest.js]: Can you explain why we need to encode and decode?
What does this fix exactly?

Comment (nimrod-becker) [reply src/endpoint/s3/s3_rest.js]: The original issue was handling the %. 
The previous PR used str replace to replace % with the encoded version.

Comment (guymguym) [reply src/endpoint/s3/s3_rest.js]: I still don't follow - how does calling `decodeURIComponent(encodeURIComponent(key))` help?
Can you provide a key example and the output?

Comment (guymguym) [reply src/endpoint/s3/s3_rest.js]: This is what I don't follow in that code:
```
> decodeURIComponent(encodeURIComponent('%'))
'%'
```

Comment (guymguym) [reply src/endpoint/s3/s3_rest.js]: ok I think I see what happened here. 
key was already decoded in https://github.com/noobaa/noobaa-core/blob/c46671f394647abaac873edea073625853c96537/src/endpoint/s3/s3_rest.js#L299
so this decode(encode(key)) and also replace(/_\$folder\$/, '/') seems to be just duplicates of what was already done in parse_bucket_and_key()...

**Pull request merged**